chalk-engine/src/logic.rs

```diff
@@ -1288,6 +1301,60 @@ impl<'forest, C: Context + 'forest, CO: ContextOps<C> + 'forest> SolveState<'for
         }
     }
 
+    /// This strand has no subgoals left, but some have floundered.
+    /// It's still possible that we can get an ambiguous answer from it though
+    fn pursue_answer_from_floundered(&mut self, strand: Strand<C>) -> Option<AnswerIndex> {
```

Comment:
I *think* we can merge this into `pursue_answer`. A good portion of these two are the same.

In `pursue_answer`, we know that if `!floundered_subgoals.is_empty()` then we should essentially have the logic of this function.